soroban-env-macros/src/synth_dispatch_host_fn_tests.rs

```diff
@@ -220,11 +220,20 @@ pub fn generate_hostfn_call_with_wrong_types(file_lit: LitStr) -> Result<TokenSt
     let mut type_to_fn_arg: BTreeMap<String, Vec<(Function, usize)>> = BTreeMap::new();
     for m in root.modules.clone() {
         for f in m.functions.clone() {
-            for (i, a) in f.args.iter().enumerate() {
-                type_to_fn_arg
-                    .entry(a.r#type.clone())
-                    .or_default()
-                    .push((f.clone(), i));
+            // checks if the current ledger protocol version falls between
```

Comment:
I'm pretty certain we'll need more protocol version checks going forward, so maybe introduce the helpers similar to ones we have in Core (i.e. protocol version before/starts/between). You actually already have duplication in this file.